flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecTemporalSort.java

```diff
@@ -104,10 +105,15 @@ public StreamExecTemporalSort(
             TimestampType keyType = (TimestampType) timeType;
             if (keyType.getKind() == TimestampKind.ROWTIME) {
                 return createSortRowTime(inputType, inputTransform, config);
-            } else if (keyType.getKind() == TimestampKind.PROCTIME) {
+            }
+        }
+        if (timeType instanceof LocalZonedTimestampType) {
+            LocalZonedTimestampType keyType = (LocalZonedTimestampType) timeType;
+            if (keyType.getKind() == TimestampKind.PROCTIME) {
```

Comment:
If `PROCTIME` will always be `LocalZonedTimestampType` from now on, shall we forbid this kind in the constructor of `ZonedTimestampType` and `TimestampType`? Would it make our life easier?